### ISSUE TITLE
Move the doc directory to separate repo

### ISSUE THREAD

**Issue by arijit79** [rust-lang/rust#76953]
The doc directory contains documentation for various tools and also submodules for various other learning resources. I think it should be moved out to be on its own separate repo. This would help documentation writers like us to clone only the docs without cloning the entire compiler sources. Plus it has become a quite large directory with many files

**Comment by est31** [comment]
Conversely, when you clone the rust repo wanting to change the compiler or standard library, cloning all the doc related submodules takes a lot of time.

**Comment by arijit79** [comment]
Exactly. Its a waste of bandwidth and storage when I just need to change 1 or 2 words in the docs

**Comment by arijit79** [comment]
I am willing to help to move the docs. But just need a confirmation that whether any member of the Rust team will work on that or shall I myself start it

**Comment by petrochenkov** [comment]
Majority of subdirectories in `src/doc` are already submodules living in separate repos (<sub>*</sub>) (see https://github.com/rust-lang/rust/tree/master/src/doc), only compiler-internal docs edited together with the compiler are living in tree.

<sub>*</sub>which is pretty annoying because you cannot simultaneously update the docs when making changes in the compiler or library, so I personally stopped contributing to docs when they were moved out.

>when I just need to change 1 or 2 words in the docs

FWIW, this can be done through GitHub interface (the "Edit this file" button) without cloning the repo.

**Comment by arijit79** [comment]
> FWIW, this can be done through GitHub interface (the "Edit this file" button) without cloning the repo.

I had already tried to do what you have suggested. I had a PR where I needed to merge 6 commits but then the assignees and the reviewers said that I needed to squash the commits. And for doing that I needed to clone the fork repo

**Comment by arijit79** [comment]
Hey guys. I just wanted to ask whether I should work on to fix this issue

**Comment by steveklabnik** [comment]
This has gone back and forth historically; there are pros and cons of each approach. When the docs were in the repo, people were concerned about bors queue length. So we moved them out, but that made coordinated changes difficult.

Any changes here would involve changing the build + deployment story, so @rust-lang/release should be at least consulted if there's a decision to change any of this.

**Comment by Mark-Simulacrum** [comment]
No, release team doesn't really care - infra might, but also not really (more or less not important whether these are submodules or not).

I think the next change to make here is probably to go through these repositories and try to move some of the larger images or artifacts to static.r-l.o or similar, there's really no reason to have them in those repositories.

**Comment by arijit79** [comment]
What is static.r.l.o. Could you please explain that to me

**Comment by petrochenkov** [comment]
If some doc subdirectory is moved to a separate repo, then it should probably use a git subtree like clippy does.
(At least from `rust-lang/rust` side we can declare the clippy subtree experiment to be a great success at this point.)

**Comment by tmandry** [comment]
Agreed with @petrochenkov, I think we should pursue a subtree approach (#70651 has more details). This gives us the best of both worlds for contributors: people who only care about docs get to clone only that repo, and people working in the compiler get to make coordinated changes easily.

**Comment by camelid** [comment]
> What is static.r.l.o. Could you please explain that to me

My understanding is that it hosts build artifacts etc. for Rust. For example, when you download a new version of Rust it is coming from static.rust-lang.org (abbreviated as static.r-l.o).

**Comment by steveklabnik** [comment]
@petrochenkov @tmandry I don't know a ton about subtree, but would be interested in hearing more about it; maybe there's some examples of how it's worked or you could ping me on zulip? that does sound like the best of both worlds, but I didn't realize it was possible.

**Comment by tmandry** [comment]
@steveklabnik [This section](https://rustc-dev-guide.rust-lang.org/contributing.html#external-dependencies-subtree) in the guide has the commands we use for clippy. I'm not too familiar with it but @oli-obk and @Manishearth probably have the most hands-on experience.

**Comment by oli-obk** [comment]
Unfortunately at this time we still need to use an unreleased version of git (details in https://github.com/rust-lang/rust-clippy/issues/5565), because the rust git repo just has too many commits and merge commits.

**Comment by arijit79** [comment]
If the subtree approach gives best of both worlds, then I am fine with that

**Comment by Manishearth** [comment]
Note that if we are doing subtrees, some team needs to commit to owning the synchronization, or we need to introduce automation for it.

**Comment by Manishearth** [comment]
Someone needs to build it and commit to running it.
